src/web/extractors.rs

```diff
@@ -584,6 +702,116 @@ mod tests {
+            .finish();
+        let result = BsoRequest::extract(&req).unwrap();
+        assert_eq!(result.user_id.legacy_id, 1);
+        assert_eq!(&result.collection, "tabs");
+        assert_eq!(&result.bso, "asdf");
+    }
+
+    #[test]
+    fn test_invalid_bso_request() {
+        let payload = HawkPayload::test_default();
+        let state = make_state();
+        let header = create_valid_hawk_header(
+            &payload,
+            &state,
+            "GET",
+            "/storage/1.5/1/storage/tabs",
+            "localhost",
+            5000,
+        );
+        let req = TestRequest::with_state(state)
+            .header("authorization", header)
+            .method(Method::GET)
+            .uri("http://localhost:5000/storage/1.5/1/storage/tabs")
+            .param("uid", "1")
+            .param("collection", INVALID_COLLECTION_NAME)
+            .finish();
+        let result = BsoRequest::extract(&req);
+        assert!(result.is_err());
+        let response: HttpResponse = result.err().unwrap().into();
+        assert_eq!(response.status(), 400);
+        let body = extract_body_as_str(&response);
+        let err: ClientRequestError = serde_json::from_str(&body).unwrap();
+        assert_eq!(err.status, 400);
+        let errors = err.errors.unwrap();
+        assert_eq!(errors[0].location, RequestErrorLocation::Path);
+    }
+
+    #[test]
+    fn test_valid_collection_request() {
+        let payload = HawkPayload::test_default();
+        let state = make_state();
+        let header = create_valid_hawk_header(
+            &payload,
+            &state,
+            "GET",
+            "/storage/1.5/1/storage/tabs",
+            "localhost",
+            5000,
+        );
+        let req = TestRequest::with_state(state)
+            .header("authorization", header)
+            .method(Method::GET)
+            .uri("http://localhost:5000/storage/1.5/1/storage/tabs")
+            .param("uid", "1")
+            .param("collection", "tabs")
+            .finish();
+        let result = CollectionRequest::extract(&req).unwrap();
+        assert_eq!(result.user_id.legacy_id, 1);
+        assert_eq!(&result.collection, "tabs");
+    }
+
+    #[test]
+    fn test_invalid_collection_request() {
+        let payload = HawkPayload::test_default();
+        let state = make_state();
+        let header = create_valid_hawk_header(
+            &payload,
+            &state,
+            "GET",
+            "/storage/1.5/1/storage/tabs",
+            "localhost",
+            5000,
+        );
+        let req = TestRequest::with_state(state)
+            .header("authorization", header)
+            .method(Method::GET)
+            .uri("http://localhost:5000/storage/1.5/1/storage/tabs")
+            .param("uid", "1")
+            .param("collection", INVALID_COLLECTION_NAME)
+            .finish();
+        let result = CollectionRequest::extract(&req);
+        assert!(result.is_err());
```

Comment:
Can save a line here and elsewhere via

`let response: HttpResponse = result.unwrap_err().into();`